Cesium3DTiles/src/Tileset.cpp

```diff
@@ -1201,12 +1197,28 @@ static bool isVisibleFromCamera(
  * @param fogDensity The fog density
  * @return Whether the tile is visible in the fog
  */
-static bool isVisibleInFog(double distance, double fogDensity) {
-  if (fogDensity <= 0.0) {
-    return true;
+static bool isVisibleInFog(
+    const std::vector<ViewState>& frustums,
+    const std::vector<double>& distances,
+    const std::vector<FogDensityAtHeight>& fogDensityTable) {
+
+  for (size_t i = 0; i < frustums.size() && i < distances.size(); ++i) {
+    const ViewState& frustum = frustums[i];
+    const double& distance = distances[i];
+
+    double fogDensity = computeFogDensity(fogDensityTable, frustum);
+
+    if (fogDensity <= 0.0) {
+      return true;
+    }
+
+    double fogScalar = distance * fogDensity;
+    if (glm::exp(-(fogScalar * fogScalar)) > 0.0) {
+      return true;
+    }
   }
-  double fogScalar = distance * fogDensity;
-  return glm::exp(-(fogScalar * fogScalar)) > 0.0;
+
+  return false;
```

Comment:
should we fog-cull a tile if it is fog-culled by all the frustums? This looks like it culls a tile if the tile is fog-culled by one of the frustum

Reply:
The logic looks correct to me. It returns true (visible) if its not in fog in any of the frustums, and returns false (culled) only if its in fog for all the frustums.

Reply:
ah that's right! Sorry for the false alarm